### PR TITLE
PWGEM:LMee: added new cuts and histogram for opening angle

### PR DESCRIPTION
**Pull request by jjungIKF**

### CODE REVIEW

File: PWGEM/Dilepton/Core/DielectronCut.h [hunk 1]
```diff
@@ -113,10 +113,25 @@ class DielectronCut : public TNamed
...
     ROOT::Math::PtEtaPhiMVector v12 = v1 + v2;
 
+    double Ptot1 = TMath::Sqrt(v1.Px() * v1.Px() + v1.Py() * v1.Py() + v1.Pz() * v1.Pz());
+    double Ptot2 = TMath::Sqrt(v2.Px() * v2.Px() + v2.Py() * v2.Py() + v2.Pz() * v2.Pz());
+
     float dca_t1_3d = dca3DinSigma(t1);
     float dca_t2_3d = dca3DinSigma(t2);
     float dca_ee_3d = std::sqrt((dca_t1_3d * dca_t1_3d + dca_t2_3d * dca_t2_3d) / 2.);
     float phiv = o2::aod::pwgem::dilepton::utils::pairutil::getPhivPair(t1.px(), t1.py(), t1.pz(), t2.px(), t2.py(), t2.pz(), t1.sign(), t2.sign(), bz);
+    float opAng = 0;
```
Comment (dsekihat) [inline PWGEM/Dilepton/Core/DielectronCut.h:123]: can you use pairutil::getOpeningAngle? This is in PairUtilities.h.

File: PWGEM/Dilepton/Core/Dilepton.h [hunk 2]
```diff
@@ -822,23 +833,38 @@ struct Dilepton {
       float deta = v1.Eta() - v2.Eta();
       float dphi = v1.Phi() - v2.Phi();
       o2::math_utils::bringToPMPi(dphi);
+      float opAng = 0;
```
Comment (dsekihat) [inline PWGEM/Dilepton/Core/Dilepton.h:836]: can you use pairutil::getOpeningAngle? This is in PairUtilities.h.

File: PWGEM/Dilepton/Core/DielectronCut.h [hunk 3]
```diff
@@ -441,8 +469,9 @@ class DielectronCut : public TNamed
   std::function<float(float)> mMaxPhivPairMeeDep{}; // max phiv as a function of mee
   bool mSelectPC{false};                            // flag to select photon conversion used in mMaxPhivPairMeeDep
   bool mApplydEtadPhi{false};                       // flag to apply deta, dphi cut between 2 tracks
-  float mMinDeltaEta{0.f};
-  float mMinDeltaPhi{0.f};
+  float mMinDeltaEta{0.f}, mMaxDeltaEta{1e10f};
```
Comment (dsekihat) [inline PWGEM/Dilepton/Core/DielectronCut.h:472]: default value of mMinDeltaEta should be -1e+10f now.

File: PWGEM/Dilepton/Core/DielectronCut.h [hunk 4]
```diff
@@ -441,8 +469,9 @@ class DielectronCut : public TNamed
...
+  float mMinDeltaPhi{0.f}, mMaxDeltaPhi{1e10f};
```
Comment (dsekihat) [inline PWGEM/Dilepton/Core/DielectronCut.h:473]: default value of mMinDeltaPhi should be -1e+10f now.

File: PWGEM/Dilepton/Core/Dilepton.h [hunk 5]
```diff
@@ -158,8 +158,12 @@ struct Dilepton {
     Configurable<float> cfg_phiv_slope{"cfg_phiv_slope", 0.0185, "slope for m vs. phiv"};
     Configurable<float> cfg_phiv_intercept{"cfg_phiv_intercept", -0.0280, "intercept for m vs. phiv"};
     Configurable<bool> cfg_apply_detadphi{"cfg_apply_detadphi", false, "flag to apply deta-dphi elliptic cut"};
-    Configurable<float> cfg_min_deta{"cfg_min_deta", 0.02, "min deta between 2 electrons (elliptic cut)"};
-    Configurable<float> cfg_min_dphi{"cfg_min_dphi", 0.02, "min dphi between 2 electrons (elliptic cut)"};
+    Configurable<float> cfg_min_deta{"cfg_min_deta", 0.02, "min deta between 2 electrons (also for elliptic cut)"};
```
Comment (dsekihat) [inline PWGEM/Dilepton/Core/Dilepton.h:161]: keep in mind that there is a boolean value "cfg_apply_detadphi" to enable deta-dphi cut.

**Comment by dsekihat** [comment]
Opening angle is equivalent to elliptic deta-dphi cut, isn't it?
The elliptic deta-dphi cut takes care of different separation power between eta and phi. The separation of opening angle along eta is better than that along phi in 2D plot (deta-dphi). In this context, your opening angle cut (omega) is included in the elliptic deta-dphi cut.

File: PWGEM/Dilepton/Core/DielectronCut.h (hunk 1 above)
Comment (jjungIKF) [reply PWGEM/Dilepton/Core/DielectronCut.h:123]: Thank you very much for pointing this out. I am sorry that I overlooked this.

File: PWGEM/Dilepton/Core/Dilepton.h (hunk 2 above)
Comment (jjungIKF) [reply PWGEM/Dilepton/Core/Dilepton.h:836]: Thank you very much for pointing this out. I am sorry that I overlooked this.

File: PWGEM/Dilepton/Core/DielectronCut.h (hunk 3 above)
Comment (jjungIKF) [reply PWGEM/Dilepton/Core/DielectronCut.h:472]: fixed

File: PWGEM/Dilepton/Core/DielectronCut.h (hunk 4 above)
Comment (jjungIKF) [reply PWGEM/Dilepton/Core/DielectronCut.h:473]: fixed

File: PWGEM/Dilepton/Core/Dilepton.h (hunk 5 above)
Comment (jjungIKF) [reply PWGEM/Dilepton/Core/Dilepton.h:161]: What do you mean by this? What should I do to account for this?

**Comment by jjungIKF** [comment]
I agree that opening angle should be somewhat included in the deta-dphi cut. But to be honest, I am missing the experience or imagination to know how exactly in translates exactly. So I would like to compare both cuts. In principle you are right and the deta-dphi should be superior as it provides more control.

File: PWGEM/Dilepton/Core/Dilepton.h (hunk 5 above)
Comment (dsekihat) [reply PWGEM/Dilepton/Core/Dilepton.h:161]: I am sorry. I overlooked if-else.

**Review by dsekihat — APPROVED** [review]

**Pull request merged**